numpy/testing/_private/utils.py

```diff
@@ -808,12 +810,7 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
 
                     # note: this definition of relative error matches that one
                     # used by assert_allclose (found in np.isclose)
-                    # Filter values where the divisor would be zero
-                    nonzero = bool_(y != 0)
-                    if all(~nonzero):
-                        max_rel_error = array(inf)
-                    else:
-                        max_rel_error = max(error[nonzero] / abs(y[nonzero]))
+                    max_rel_error = np.max(error / abs(y))
```

Comment:
I don't understand this change. It will overflow. Perhaps the idea was to use `np.max` in the original code one line 816 and something got messed up in a merge?

Reply:
reverted this change and added `np.` to `max`, `abs`. Rebased against latest HEAD. Tests seem to pass now.